src/lib/customPropTypes.js

```diff
@@ -12,63 +12,80 @@ export const as = (...args) => PropTypes.oneOfType([
   PropTypes.func,
 ])(...args)
 
-/* eslint-disable max-nested-callbacks */
-const findBestSuggestions = _.memoize((propValueWords, suggestions) => _.flow(
-  _.map((suggestion) => {
-    const suggestionWords = suggestion.split(' ')
-
-    const propValueScore = _.flow(
-      _.map(x => _.map(y => leven(x, y), suggestionWords)),
-      _.map(_.min),
-      _.sum,
-    )(propValueWords)
-
-    const suggestionScore = _.flow(
-      _.map(x => _.map(y => leven(x, y), propValueWords)),
-      _.map(_.min),
-      _.sum,
-    )(suggestionWords)
-
-    return { suggestion, score: propValueScore + suggestionScore }
-  }),
-  _.sortBy(['score', 'suggestion']),
-  _.take(3),
-)(suggestions))
-/* eslint-enable max-nested-callbacks */
 
 /**
  * Similar to PropTypes.oneOf but shows closest matches.
  * Word order is ignored allowing `left chevron` to match `chevron left`.
  * Useful for very large lists of options (e.g. Icon name, Flag name, etc.)
  * @param {string[]} suggestions An array of allowed values.
  */
-export const suggest = suggestions => (props, propName, componentName) => {
+export const suggest = (suggestions) => {
   if (!Array.isArray(suggestions)) {
-    throw new Error([
-      'Invalid argument supplied to suggest, expected an instance of array.',
-      ` See \`${propName}\` prop in \`${componentName}\`.`,
-    ].join(''))
+    throw new Error('Invalid argument supplied to suggest, expected an instance of array.')
   }
-  const propValue = props[propName]
 
-  // skip if prop is undefined or is included in the suggestions
-  if (_.isNil(propValue) || propValue === false || _.includes(propValue, suggestions)) return
+  /* eslint-disable max-nested-callbacks */
+  const findBestSuggestions = _.memoize((str) => {
```

Comment:
Implementation is the same, just passed the base string in instead of an array so that `_.memoize` can actually do its job, since arrays are not memoized by value:

```js
let calls = 0
var fn = _.memoize((arr) => ++calls)

fn(['my', 'words'])
fn(['my', 'words'])
console.log(calls) // 2
```

Note that the previous usage of this memoized function was incorrect, since it would ignore the second argument `suggestions`, meaning the first call to it with a given propValue would have won out. This was unintentionally mitigated by the fact that `_.memoize` only tracks the first argument by default, which meant that the memoization was effect just a noop.

The _real_ fix would be to write our own resolver function for the memoization, but it would essentially achieve the same thing as I'm doing here by inlining the function.